### PR TITLE
SYM-99 : Adding error handling and logging to Rest Api.

### PR DESCRIPTION
**Pull request by mvu02**
SYM-99 : Adding error handling and logging to Rest Api.

### CODE REVIEW

**Comment by maoo** [comment]
Please do not merge this PR until we validate CLA from @mvu02 . Thank you!

**Comment by maoo** [comment]
@mvu02 was added to our whitelists, thanks @malaysf for providing info! The Pull Request can be safely merged now.

File: src/SymphonyOSS.RestApiClient/Api/AuthenticationApi/AppAuthenticationApi.cs [hunk 1]
```diff
@@ -39,18 +45,35 @@ public X509Certificate2 GetPodCertificate()
             }
             catch (Exception e)
             {
+                _log.LogDebug(0, e, "An error has occured while trying to retrieve the pod certifacte.");
```
Comment (malaysf) [inline src/SymphonyOSS.RestApiClient/Api/AuthenticationApi/AppAuthenticationApi.cs]: certificate is misspelled

**Comment by malaysf** [comment]
Some of the exceptions are logged as error messages, some as debug messages. What is the distinction when using one over the other?

**Comment by mvu02** [comment]
I was trying to follow the style of logging that was already present. I saw that the debug messages were sometimes used as informational and for some exceptions that didn't seem catastrophic, and I log error messages for api calls that seems like it would be important if it were to fail. Should I be logging all exceptions as LogError?

File: src/SymphonyOSS.RestApiClient/Api/AuthenticationApi/AppAuthenticationApi.cs (hunk 1 above)
Comment (mvu02) [reply src/SymphonyOSS.RestApiClient/Api/AuthenticationApi/AppAuthenticationApi.cs]: Got it thanks!

**Comment by mvu02** [comment]
Hey @malaysf , I have tested my changes.

**Comment by malaysf** [comment]
@mvu02 I'll get this accepted and built once I get back next week.

**Comment by malaysf** [comment]
@mvu02 All the places where you have something like:
`_log.LogError`

need to be replaced with
`_log?.LogError`

There may be no logger factory installed, in which case _log will be null.

**Comment by mvu02** [comment]
Hi @malaysf, that makes sense. I have made the changes and amended the commit. As per our offline conversation, for future PRs I will be doing separate commits so it's easier to review the on-going changes. Thanks!

**Pull request merged**